feat(shell-integration): add automatic integration for Elvish

ulzu opened the pull request:
Depends on PR #1772.

Fish automatic integration taken as an example.
Just like Fish, Elvish checks `XDG_DATA_DIRS` for its modules.
Thus, Fish integration in zig is reused, and integration in
Elvish now removes `GHOSTTY_FISH_XDG_DIR` environment variable
on launch.

Commit:
feat(shell-integration): add automatic integration for Elvish

Fish automatic integration taken as an example.
Just like fish, Elvish checks `XDG_DATA_DIRS` for its modules.
Thus, Fish integration in zig is reused, and integration in
Elvish now removes `GHOSTTY_FISH_XDG_DIR` environment variable
on launch.

jparise reviewed (commented):
Aside from the minor ordering (alphabetization) note, the `shell_integration` bits look simple-enough to me. Maybe we want to generalize the `XDG_DATA_DIRS` code into a common function instead of having `setupElvish` call `setupFish`?

Lastly, could you also update the main `README.md`?

Commit:
refactor(shell-integration): refactor to make cases alphabetical

also fix: build errors
also fix: name mismatch (GHOSTTY_FISH_XDG_DIR -> GHOSTTY_INTEGRATION_DIR)
also refactor elvish file to evade unobvious returns

Commit:
refactor(shell-integration): refactor to make cases alphabetical

also refactors elvish file to evade unobvious returns
and tries to fix some build errors

mitchellh committed:
prettier

mitchellh committed:
rename env var
